client-implementation.md

```diff
@@ -0,0 +1,67 @@
+# Client Implementations
+
+This document describes client implementation recommendations for interoperability with registry servers and other clients using the OCI Distribution Spec.
+
+## Backwards Compatibility
+
+Client implementations SHOULD support registries that implement partial or older versions of the OCI Distribution Spec.
+This section describes how client fallback procedures when an API is not available.
+
+### Referrers API
+
+The Referrers API here is described by [Listing Referrers](spec.md#listing-referrers) and [end-12a](spec.md#endpoints).
+
+A client that pushes an Image or Artifact manifest with a defined `Refers` field MUST verify the Referrers API is available.
+A client querying the Referrers API and receiving a 404 MUST fallback to using an Index pushed to a tag described by the following schema.
+
+#### Referrers Tag Schema
+
+```text
+<alg>-<ref>
+```
+
+- `<alg>`: the digest algorithm (e.g. `sha256` or `sha512`)
+- `<ref>`: the digest from the `refers` field (limit of 64 characters)
+
+For example, a manifest with the `Refers` field digest set to `sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` in the `registry.example.org/project` repository would have a expect an Index at `registry.example.org/project:sha256-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
+
+#### Pushing Manifests with Refers
+
+When pushing an Image or Artifact manifest with the `Refers` field and the Referrers API returns a 404, the client MUST:
+
+1. Pull the Index using the tag schema.
+1. If the tag returns a manifest other than the expected Index, the client SHOULD report a failure and skip the remaining steps.
```

Comment:
```suggestion
1. If the pull manifest by digest returns a manifest other than the expected manifest, the client SHOULD report a failure and skip the remaining steps.
```

Reply:
similar changes below...

Reply:
The pull in this case is for a tag, not a digest. The tag schema just happens to contain the digest of a different manifest.

Reply:
yeah sort of confused here given we just defined referrers tag schema.. then proceed to provide a push manifest example (image or artifact) and now we are pulling the index using the tag schema.. I read it as meaning the referrers tag schema)  

hard to tell in this example if there is an image.index to pull yet by tag and if there is one what does that index have to do with the image/arfifact manifest we are pushing as that would be a different image.index? 

maybe more examples, or more explanation?

Reply:
The phrasing has been changed to say "Referrer list" rather than "Index" to hopefully make this one clearer.